pkg/util/acrtoken/acrtoken.go

```diff
@@ -142,6 +152,33 @@ func (m *manager) RotateTokenPassword(ctx context.Context, rp *api.RegistryProfi
 	return nil
 }
 
+// ValidateToken attempts to verify that the current ACR token functions
+func (m *manager) ValidateToken(ctx context.Context, rp *api.RegistryProfile) error {
+	creds := clusterauthorizer.Credentials{
+		ClientID:     []byte(rp.Username),
+		ClientSecret: []byte(rp.Password),
```

Comment:
It looks like you're using the registry profile that's stored on the cluster doc in CosmosDB to get the token password. I chose not to even worry about updating that password during the rotation process because the only time the cluster cares about that password during the initial cluster installation — that's how the RP passes the username (token name) and password to the cluster during the bootstrap process.

The cluster's "source of truth" for its current ACR password is the `cluster` secret in `openshift-azure-operator`. That's what it uses during the pullsecret controller's reconciliation to ultimately construct the pull secret that nodes use to get container images. The cluster never accesses the password on the cluster document after installation, so I opted not to as well in order to retain a single source of truth for the password.

As far as testing the validity of the new credentials, it would make sense to me to use the exact same credentials the cluster uses in order to minimize potential differences in data. That means getting the secret from the cluster instead of the RP, and deserializing the token stored on it.